engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/system/CreateProcessNode.java

```diff
@@ -0,0 +1,115 @@
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.Language;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.Context;
+import org.enso.interpreter.runtime.data.Vector;
+import org.enso.interpreter.runtime.error.PanicException;
+
+import java.io.*;
+
+@BuiltinMethod(
+    type = "System",
+    name = "create_process",
+    description = "Create a system process, returning the exit code.")
+public abstract class CreateProcessNode extends Node {
+
+  static CreateProcessNode build() {
+    return CreateProcessNodeGen.create();
+  }
+
+  abstract Object execute(
+      Object _this,
+      String command,
+      Vector arguments,
+      String input,
+      boolean redirectIn,
+      boolean redirectOut,
+      boolean redirectErr);
+
+  @Specialization
+  @CompilerDirectives.TruffleBoundary
+  Object doCreate(
+      Object _this,
+      String command,
+      Vector arguments,
+      String input,
+      boolean redirectIn,
+      boolean redirectOut,
+      boolean redirectErr,
+      @CachedContext(Language.class) Context ctx) {
+    String[] cmd = new String[(int) arguments.getArraySize() + 1];
+    cmd[0] = command;
+    System.arraycopy(arguments.getItems(), 0, cmd, 1, (int) arguments.getArraySize());
+    TruffleProcessBuilder pb = ctx.getEnvironment().newProcessBuilder(cmd);
+
+    try {
+      Process p = pb.start();
+      ByteArrayInputStream in = new ByteArrayInputStream(input.getBytes());
+      ByteArrayOutputStream out = new ByteArrayOutputStream();
+      ByteArrayOutputStream err = new ByteArrayOutputStream();
+
+      try (OutputStream processin = p.getOutputStream()) {
+        InputStream stdin;
+        if (redirectIn) {
+          stdin = ctx.getIn();
+        } else {
+          stdin = in;
+        }
+        int nread;
+        byte[] buf = new byte[8096];
+        while (stdin.available() > 0 && (nread = stdin.read(buf)) != -1) {
+          processin.write(buf, 0, nread);
+        }
+      } catch (IOException ignored) {
+        // Getting the output stream of a finished process results in an IOException.
+        // We can ignore it at this point.
+      }
+
+      p.waitFor();
+
+      try (InputStream processout = p.getInputStream()) {
+        OutputStream stdout;
+        if (redirectOut) {
+          stdout = ctx.getOut();
+        } else {
+          stdout = out;
+        }
+        int nread;
+        byte[] buf = new byte[8096];
+        while ((nread = processout.read(buf)) != -1) {
+          stdout.write(buf, 0, nread);
+        }
+      }
+
+      try (InputStream processerr = p.getErrorStream()) {
+        OutputStream stderr;
+        if (redirectErr) {
+          stderr = ctx.getErr();
+        } else {
+          stderr = err;
+        }
+        int nread;
+        byte[] buf = new byte[8096];
+        while ((nread = processerr.read(buf)) != -1) {
+          stderr.write(buf, 0, nread);
+        }
+      }
+
+      long exitCode = p.exitValue();
```

Comment:
Is exit codes as longs the abstraction we really want? In a language like Enso we'd rather a sum type `ExitSuccess | ExitFailure value | ...`.

Reply:
Fixed in stdlib